test-suite/unit-tests/toplevel/proof_diffs_html_test.ml

```diff
@@ -0,0 +1,62 @@
+open OUnit
+open Utest
+open Proof_diffs_html
+
+let tests = ref []
+let add_test name test = tests := (mk_test name (TestCase test)) :: !tests
+let cfprintf oc = Printf.(kfprintf (fun oc -> fprintf oc "") oc)
+let log_out_ch = ref stdout
+let cprintf s = cfprintf !log_out_ch s
+let string_of_string s : string = "\"" ^ s ^ "\""
+
+let t () =
+  (* set up printing into log file *)
+  (log_out_ch :=  match !Utest.log_out_ch with
+                 | Some ch -> ch
+                 | None -> stdout);
+  Proof_diffs.log_out_ch := !log_out_ch
+let _ = add_test "setup" t
+
+
+(* todo: other tests to write
+write_html_diffs BLOCKED: CAN'T GET TAGS IN BUFFER FORMATTER
+  has right tags
+  splits at end of line
+*)
+
+
+
+let t () =
+  let orig = "V&W<>X\"Y\'Z" in
+  let escaped = maybe_escape orig true in
+  let expected = "V&amp;W&lt;&gt;X&quot;Y&apos;Z" in
+
+  assert_equal ~msg:"escapes added" ~printer:string_of_string expected escaped;
+  let not_escaped = maybe_escape orig false in
+  assert_equal ~msg:"escapes not added" ~printer:string_of_string orig not_escaped
+let _ = add_test "maybe_escape" t
+
+
+(* todo: not getting any tags from buffer formatter!!! *)
+(*
+let t () =
+  let o_pp = seq [] in
```

Comment:
I suppose you intend to get rid of these todos and commented parts before merging.

Reply:
I'm generally very doubtful about the interest of keeping commented out code. It is bad for maintainability. If that's for documentation purposes, there should be more text than code. If that's for later improvement, I would say: keep it in a private branch. Or create a new issue listing further possible improvements and post the code there.